src/function/vector_cast_functions.cpp

```diff
@@ -154,28 +154,24 @@ static void nestedTypesCastExecFunction(
 
 bool CastFunction::hasImplicitCast(const LogicalType& srcType, const LogicalType& dstType) {
     // We allow cast between any numerical types
+    // double check here TODO: should remove it after correctly get cast cost
```

Comment:
I agree we can probably remove `hasImplicitCast()` and replace it with `getCastCost`

Reply:
There are other cases where we need hasImplicitCast().
She may have to go through them one by one.
e.g.
```
std::unique_ptr<BoundReadingClause> Binder::bindUnwindClause(const ReadingClause& readingClause) {
    auto& unwindClause = ku_dynamic_cast<const ReadingClause&, const UnwindClause&>(readingClause);
    auto boundExpression = expressionBinder.bindExpression(*unwindClause.getExpression());
    boundExpression =
        ExpressionBinder::implicitCastIfNecessary(boundExpression, LogicalTypeID::VAR_LIST);
    auto aliasExpression = createVariable(
        unwindClause.getAlias(), *VarListType::getChildType(&boundExpression->dataType));
    return make_unique<BoundUnwindClause>(std::move(boundExpression), std::move(aliasExpression));
}
```
@andyfengHKU You may have to point out how to refactor these functions as well.